### ISSUE TITLE
Bind observable to collection of observers

### ISSUE THREAD

**Issue by sgtsquiggs** [RxSwiftCommunity/RxSwiftExt#163]
### Name and description

An extension on ObservableType to allow `.bind(to:)` with a collection of observers, and an extension on Collection (where Iterator.Element == Disposable) to allow `.disposed(by:)` with a collection of Disposables.

### Motivation for inclusion

Useful when you have a single observable binding to a lot of observers. A big use case is when implementing forms and having a single observable bind to isEnabled on a multitude of controls.

### Example of use

```swift
isEnabledSubject
  .bind(to: myCollectionOfTextFields.map({$0.rx.isEnabled}))
  .disposed(by: disposeBag)
```

Implementation here: https://gist.github.com/sgtsquiggs/0b1dc9dbd6fe51f107d4082ac7c448cd

**Comment by jegnux** [comment]
I agree with @freak4pc 
It's a an interesting addition, but we should indeed use `CompositeDisposable` instead. It even simplifies the implementation to only that:

```swift
extension ObservableType {
    func bind<O>(to observers: [O]) -> Disposable where O : ObserverType, Self.E == O.E {
        let shared = self.share()
        let disposables = observers.map(shared.bind(to:))
        return CompositeDisposable(disposables: disposables)
    }
}
```

**Comment by valerianb** [comment]
There was recently a PR on RxSwift to take care of disposables in one call:
https://github.com/ReactiveX/RxSwift/pull/1676

**Comment by sgtsquiggs** [comment]
Oh I wasn't aware of CompositeDisposable. I'll update my gist!

**Comment by freak4pc** [comment]
@valerianb that is also a cool PR (the `.bag` one) but doesn't solve the same thing exactly IMO. 

You would still need to repeat the binding (a.bind(to: b), a.bind(to:c)), instead of a.bind(to: [b, c])). So when thinking about this the last few days I think this could be a great addition if you'd like to build it @sgtsquiggs

**Comment by MortyMerr** [comment]
Closing, cause issue was resolved with RxSwift 5.0
